modules/component-store/src/component-store.ts

```diff
@@ -156,6 +156,29 @@ export class ComponentStore<T extends object> implements OnDestroy {
     }
   }
 
+  /**
+   * Patches the state with provided partial state.
+   *
+   * @param partialStateOrUpdaterFn a partial state or a partial updater
+   * function that accepts the state and returns the partial state.
+   * @throws Error if the state is not initialized.
+   */
+  patchState(
+    partialStateOrUpdaterFn: Partial<T> | ((state: T) => Partial<T>)
+  ): void {
+    if (!this.isInitialized) {
+      throw new Error(this.notInitializedErrorMessage);
+    }
+
+    const updaterFn = (state: T) => ({
```

Comment:
a few suggestions:
1. We can use `this.setState` here
2. Let's extract and evaluate `patchedState` separately for readability purposes.

```suggestion
    this.setState((state) => {
      const patchedState = (typeof partialStateOrUpdaterFn === 'function') 
          ? partialStateOrUpdaterFn(state)
          : partialStateOrUpdaterFn;
      return {
      ...state,
      ...patchedState, // 👈 comes after `state` to override the properties
      };
    }
```

Reply:
done